src/windowmanagement/Window/Window.tsx

```diff
@@ -0,0 +1,33 @@
+import { PropsWithChildren } from 'react';
+import { ErrorBoundary } from 'react-error-boundary';
+import { useDispatch } from 'react-redux';
+import { ScrollArea } from '@mantine/core';
+import { useElementSize } from '@mantine/hooks';
+
+import { reloadPropertyTree } from '@/redux/propertytree/propertyTreeMiddleware';
+import { fallbackRender } from '@/util/fallbackRenderer';
+
+import { WindowSizeContext } from './WindowSizeContext';
+
+export function Window({ children }: PropsWithChildren) {
+  const { ref, width, height } = useElementSize();
+  const dispatch = useDispatch();
+  return (
+    <WindowSizeContext.Provider
+      value={{
+        width: width,
+        height: height
+      }}
+    >
+      <ScrollArea h={'100%'} ref={ref}>
+        <ErrorBoundary
+          fallbackRender={fallbackRender}
+          onReset={() => dispatch(reloadPropertyTree())}
```

Comment:
I don't understand the choice of reset function here, it should be general in this case, no? If so, something like a window.reload would be sufficient or if we can come up with some smarter solution to try and reload a specific component

Reply:
I just put something there was something, not a lot of thought put into it:) I'll just make a window reload and a todo comment